### PR TITLE
CRM-20532 Fix issue where cannot get information on a single extensio…

### PR DESCRIPTION
**Pull request by seamuslee001**
…n with extension.get api and ensure that Fields specified in return key are supported

### CODE REVIEW

**Comment by seamuslee001** [comment]
ping @eileenmcnaughton @twomice @ineffyble I think this fixes it in an elegant way whilst still supporting that effort to have all information for Extensions returned

File: tests/phpunit/api/v3/ExtensionTest.php [hunk 1]
```diff
@@ -57,4 +57,35 @@ public function testGetremote() {
     $this->assertEquals('CiviDiscount', $result['values'][0]['name']);
   }
 
+  /**
+   * Test retunging a single extension
+   */
+  public function testSingleExtesnionGet() {
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/ExtensionTest.php]: lysdexia rules

File: api/v3/Extension.php [hunk 2]
```diff
@@ -332,6 +332,7 @@ function _civicrm_api3_extension_refresh_spec(&$fields) {
  *   API result
  */
 function civicrm_api3_extension_get($params) {
+  $keys = (array) $params['key'];
```
Comment (eileenmcnaughton) [inline api/v3/Extension.php]: could this cause an enotice?

Comment (seamuslee001) [inline api/v3/Extension.php]: I think you can cast a null as an array

Comment (eileenmcnaughton) [inline api/v3/Extension.php]: would it never not be set?

File: api/v3/Extension.php [hunk 3]
```diff
@@ -346,9 +347,17 @@ function civicrm_api3_extension_get($params) {
     }
     $info = CRM_Extension_System::createExtendedInfo($obj);
     $info['id'] = $id++; // backward compatibility with indexing scheme
-    $result[] = $info;
+    if (!empty($params['key'])) {
```
Comment (eileenmcnaughton) [inline api/v3/Extension.php]: what about moving the filtering to the start of the loop - I have a feeling some of those calls are expensive

Comment (seamuslee001) [inline api/v3/Extension.php]: I tried that and didn't seem to get anywhere

Comment (seamuslee001) [inline api/v3/Extension.php]: Also doing it further down keeps the indexing the same no matter what the get is like

Comment (eileenmcnaughton) [inline api/v3/Extension.php]: ok - is this not slow? $info = CRM_Extension_System::createExtendedInfo($obj);

Comment (seamuslee001) [inline api/v3/Extension.php]: Doesn't feel that slow to me

Comment (eileenmcnaughton) [inline api/v3/Extension.php]: ok

**Comment by eileenmcnaughton** [comment]
Makes sense - if the tests can't find anything wrong then I can't either

**Comment by colemanw** [comment]
@eileenmcnaughton ought the jira ticket https://issues.civicrm.org/jira/browse/CRM-20532 be closed as well?

**Comment by seamuslee001** [comment]
@colemanw I'm dealing with JIRA and I think I'll put a note on the API change log

**Pull request merged**